presto-main/src/main/java/com/facebook/presto/cost/AggregationStatsRule.java

```diff
@@ -53,17 +58,31 @@ protected Optional<PlanNodeStatsEstimate> doCalculate(AggregationNode node, Stat
             return Optional.empty();
         }
 
-        if (node.getStep() != SINGLE) {
```

Comment:
why did you remove this? e.g. partial aggregation output should probably be estimated to be the same as source stats.  But in any case, the node step only shows up in the final plan.  At the time when we are making cost based decisions, the only step we see is single

Reply:
I remove this logic so that it can sort of "propagate" the stats calculated during SINGLE step phase to the double agg phase where plan will show up as having missing estimated stats. In other words, if we keep this logic, the aggregation nodes with PARTIAL step and FINAL step would return NaN stats causing confusing as if the aggregation node has an unknown estimates but it should have given some stats estimation during SINGLE step phase. Do you have any suggestion on how to avoid this? because current cost model derived from stats has this issue where `CostCalculatorUsingExchanges` cannot derive cost from PARTIAL step which returns unknown cost instead https://github.com/prestodb/presto/blob/a90ff6b43f4acfd0be1f0d13c28a355138e7e79f/presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java#L168